src/engraving/libmscore/beam.cpp

```diff
@@ -1222,7 +1254,12 @@ void Beam::offsetBeamWithAnchorShortening(std::vector<ChordRest*> chordRests, in
     int newDictator = dictator;
     int newPointer = pointer;
     int reduce = 0;
-    while (!isValidBeamPosition(newDictator, isStartDictator, isAscending, isFlat, staffLines, true)) {
+    auto fourBeamException = [staffLines](int beams, int yPos) {
+        yPos += 400;
```

Comment:
What does this number mean?

Reply:
There are a lot of new magic numbers in this PR. You should at least add a comment to each of them

Reply:
I've added a comment to that line, and I've searched for other places where the numbers aren't immediately obvious and couldn't find other ones. The rest make sense with the understanding that all of the beam layout logic uses units of 0.25sp.